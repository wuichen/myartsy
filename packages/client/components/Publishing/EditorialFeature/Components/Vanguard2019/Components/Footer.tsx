import { Box, Flex, Sans, Separator, Serif } from "@artsy/palette"
import React from "react"

export const VanguardFooter = () => {
  return (
    <Box background="white" pb={80}>
      <Box
        maxWidth={["100vw", 1000]}
        px={["10vw", "10vw", "10vw", "10vw", 4]}
        mx="auto"
      >
        <Separator mb={4} />
        <Flex flexDirection={["column", "row"]}>
          <Box pr={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Corrections
            </Sans>
            <Serif size="3" element="p">
              {corrections}
            </Serif>
          </Box>
          <Box pl={[0, 2]} width={["100%", "50%"]} pb={4}>
            <Sans size="3" weight="medium">
              Clarifications
            </Sans>
            <Serif size="3" element="p">
              {clarifications}
            </Serif>
          </Box>
        </Flex>
        <Separator mb={2} />
        <Serif size="3">
          Back to <a href="/articles">Artsy Editorial</a>
        </Serif>
      </Box>
    </Box>
  )
}

const corrections = `A previous version of this article referred to Aike gallery as
Aike-Dellarco; McArthur Binion’s show at the Mississippi Museum of
Art, originally scheduled for this fall, is not opening until 2020;
and Melike Kara is Kurdish-German not Turkish-German. The text has
been updated to reflect these changes.`

const clarifications = `While only Beyoncé was named as the buyer of a Derek Fordjour work at
Frieze New York, Jay-Z was actually the buyer; in order to fully
recognize the breadth of galleries that represent the featured
artists, additional gallery names have been added to the text for
Beatriz González, Vivian Suter, and Diane Simpson; while the text on
Henry Taylor previously referred to Eva Presenhuber as his dealer, it
has been updated to reflect that Blum & Poe has been primarily
representing the artist for the past decade; and mention of
SculptureCenter, organizers of Tishan Hsu’s 2020 show at the Hammer,
has been added to clarify their involvement with the exhibition.`
