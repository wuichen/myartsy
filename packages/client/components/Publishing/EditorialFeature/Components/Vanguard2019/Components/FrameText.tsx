import { color, Flex, Sans } from "@artsy/palette"
import React from "react"
import styled from "styled-components"
import { Media } from "Utils/Responsive"

interface VanguardFrameTextProps {
  isSlideOpen?: boolean
}

export const VanguardFrameText: React.SFC<VanguardFrameTextProps> = props => {
  const { isSlideOpen } = props

  return (
    <>
      <Media at="xs">
        <MobileFrame isSlideOpen={isSlideOpen}>
          <MobileFrameText size="8">The Artsy</MobileFrameText>
          <MobileFrameText size="8" isUppercase>
            Vanguard
          </MobileFrameText>
          <MobileFrameText size="8" isUppercase>
            2019
          </MobileFrameText>
        </MobileFrame>
      </Media>

      {/* Desktop */}
      <Media greaterThan="xs">
        <FrameTextLeft
          isSlideOpen={isSlideOpen}
          size={["12", "12", "14", "16"]}
        >
          Vanguard
        </FrameTextLeft>
        <FrameTextRight
          isSlideOpen={isSlideOpen}
          size={["12", "12", "14", "16"]}
        >
          2019
        </FrameTextRight>
      </Media>
    </>
  )
}

const FrameText = styled(Sans)`
  position: fixed;
  top: 50%;
  text-transform: uppercase;
  z-index: 2;
  mix-blend-mode: difference;
  color: ${color("white100")};
  will-change: color;
`

const FrameTextLeft = styled(FrameText)<{ isSlideOpen?: boolean }>`
  left: 0;
  transform: rotate(-90deg);
  transform-origin: 30% 136%;

  &&& {
    z-index: ${p => (p.isSlideOpen ? 0 : 2)};
  }
`

const FrameTextRight = styled(FrameText)<{ isSlideOpen?: boolean }>`
  right: 0;
  transform: rotate(90deg);
  transform-origin: 60% 60%;

  &&& {
    z-index: ${p => (p.isSlideOpen ? 0 : 2)};
  }
`

const MobileFrame = styled(Flex)<{
  isSlideOpen?: boolean
}>`
  position: fixed;
  flex-direction: column;
  top: 65px;
  left: -10px;
  height: calc(100vh - 150px);
  width: 60px;
  justify-content: space-between;
  z-index: ${p => (p.isSlideOpen ? 0 : 2)};
`

const MobileFrameText = styled(Sans)<{
  isUppercase?: boolean
}>`
  transform: rotate(90deg);
  white-space: nowrap;
  padding: 0 10px;

  ${({ isUppercase }) =>
    isUppercase &&
    `
    text-transform: uppercase;
  `}
`
