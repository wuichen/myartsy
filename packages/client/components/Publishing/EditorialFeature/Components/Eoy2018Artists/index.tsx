import { Box, color, Flex, media, Sans, Serif, space } from "@artsy/palette"
import { compact, find, map } from "lodash"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { EditorialFeaturesProps } from "Components/Publishing/EditorialFeature/EditorialFeature"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import { ArticleCardContainer } from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCard"
import {
  ArticleCards,
  ArticlesWrapper,
} from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCards"
import {
  FullLabel,
  ImageSetContainer,
  ImageSetPreview,
  ImgContainer,
} from "Components/Publishing/Sections/ImageSetPreview"
import {
  LabelWrapper,
  SlideshowCta,
  SlideshowTitle,
} from "Components/Publishing/Sections/ImageSetPreview/ImageSetLabel"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { Text } from "Components/Publishing/Sections/Text"
import { resize } from "Utils/resizer"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export class Eoy2018Artists extends React.Component<EditorialFeaturesProps> {
  getHeaderSections = () => {
    const { sections } = this.props.article
    const headers = []

    sections.map((section, index) => {
      let headerSection
      const sectionAfter = sections[index + 1]
      const sectionAfterisImage =
        sectionAfter && sectionAfter.type === "image_collection"

      if (this.sectionIsHeader(section)) {
        headerSection = {
          index,
          section,
        }
        if (sectionAfterisImage) {
          headerSection.imageSection = sectionAfter
        }
        headers.push(headerSection)
      }
    })
    return headers
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionArtistHeader = (section, i) => {
    const headerSections = this.getHeaderSections()
    const { imageSection } = find(headerSections, ["section", section])
    const src = imageSection && imageSection.images[0].url
    const caption = imageSection && imageSection.images[0].caption

    return (
      <ArtistHeaderSection
        key={i}
        mb={40}
        flexDirection={["column", "column", "row", "row"]}
      >
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg
          src={src && resize(src, { width: 700 })}
          alignItems="flex-end"
        >
          {caption && (
            <Sans size="1" color="white" px={10} py={5}>
              <div dangerouslySetInnerHTML={{ __html: caption }} />
            </Sans>
          )}
        </ArtistHeaderImg>
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    const {
      article: { sections, layout },
      showTooltips,
    } = this.props
    const isChapterStart =
      sections[i - 1] && sections[i - 1].type === "image_collection"

    return (
      <Box maxWidth={["100%", "75%", "75%"]} ml="auto" px={[20, 0]} key={i}>
        <TextSection size="5" pb={[40, 60]} isChapterStart={isChapterStart}>
          <Text
            html={section.body}
            layout={layout}
            showTooltips={showTooltips}
          />
        </TextSection>
      </Box>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper
        key={i}
        mb={60}
        maxWidth={["100%", "100%", "100%", "75%"]}
      >
        <ImageSetPreview section={section}>
          <CaptionWrapper size={["3", "4"]}>
            <ImageSetCaption
              dangerouslySetInnerHTML={{ __html: section.images[0].caption }}
            />
          </CaptionWrapper>
        </ImageSetPreview>
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props
    const headerSections = this.getHeaderSections()
    const headerTextIndexes = map(headerSections, "index")
    const headerImages = map(headerSections, "image")

    const renderedSections = sections.map((section, i) => {
      const isHeader = headerTextIndexes.includes(i)
      const isHeaderImage = headerImages.includes(section)

      if (isHeader) {
        return this.sectionArtistHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text" && i !== 0) {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    const { article, isMobile, isTablet, isTest } = this.props
    const introText = this.sectionText(article.sections[0], 0)
    const headerImages = map(
      compact(map(this.getHeaderSections(), "imageSection")),
      "images"
    )

    return (
      <AricleWrapper>
        <Nav canFix color={color("black100")} backgroundColor="white">
          <NavBorder />
        </Nav>

        <Box px={[10, 10, 55]} maxWidth={1600} mx="auto">
          <Eoy2018ArticleHeader
            images={headerImages}
            isMobile={isMobile}
            isTablet={isTablet}
            isTest={isTest}
          />

          <ArticleContent py={40}>
            <IntroSection
              alignItems="flex-start"
              flexDirection={["column", "row"]}
              pl={[0, 20]}
            >
              <Byline article={article} />
              {introText}
            </IntroSection>
            {this.getSections()}
          </ArticleContent>
        </Box>

        {article.relatedArticles && (
          <Box px={[10, 10, 55]} mb={40} mx="auto">
            <RelatedArticleWrapper>
              <ArticleCards relatedArticles={article.relatedArticles} />
            </RelatedArticleWrapper>
          </Box>
        )}
      </AricleWrapper>
    )
  }
}

const BORDER_WIDTH = 6

const AricleWrapper = styled.div`
  ${NavContainer} {
    border-bottom: none;
  }
`

const NavBorder = styled.div`
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  position: absolute;
  top: 100%;
  left: 55px;
  right: 55px;
  max-width: calc(1600px - 110px);
  margin: 0 auto;

  ${media.md`
    left: 10px;
    right: 10px;
  `};
`

const ArticleContent = styled(Box)`
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
    line-height: 1.3em;
    margin: 0;
    font-weight: inherit;

    ${media.sm`
      ${unica("s25")};
      line-height: 1.3em;
    `};
  }
`

export const ArtistHeaderSection = styled(Flex)`
  height: 60vh;
  min-height: 450px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left-width: 0;

  ${media.md`
    height: fit-content;
  `};
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  overflow: hidden;
  min-height: min-content;

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    min-height: fit-content;
    padding: ${space(2)}px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    margin: 0;
    font-weight: inherit;

    ${media.md`
      width: 60%;
      float: right;
      height: 100%;
      border-bottom: none;
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};

    ${media.xs`
      ${unica("s45")};
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
      border-left: none;
      width: 100%;
      height: 50%;
      float: none;
    `};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    min-height: fit-content;
    display: inline-flex;
    padding: ${space(2)}px;
    margin: 0;
    font-weight: inherit;

    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }

    ${media.md`
      width: 40%;
      float: left;
      &:last-child {
        border-left: none;
        border-top: ${BORDER_WIDTH}px solid ${color("purple100")};
      }
    `};

    ${media.xs`
      ${unica("s19")};
      width: 50%;
      float: none;
      &:last-child {
        border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
        border-top: none;
      }
    `};
  }

  ${media.md`
    height: 65vh;
    border-right: none;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ArtistHeaderImg = styled(Flex)<{ src?: string }>`
  flex: 1;
  background: ${color("purple100")};

  p {
    text-shadow: 0 0 5px black;
    opacity: 0.6;
  }

  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};

  ${media.md`
    min-height: 70vw;
  `};
`

const IntroSection = styled(Flex)`
  ${BylineContainer} {
    flex-direction: column;
    align-items: flex-start;
    ${media.md`
      padding: 0 20px 20px;
    `};
  }
`

const TextSection = styled(Serif)<{ isChapterStart?: boolean }>`
  ${StyledText} {
    blockquote {
      padding: 0;
    }

    p,
    .paragraph {
      font-size: 24px;
      text-indent: 2em;
      padding: 0;

      ${props =>
        props.isChapterStart &&
        `
        &:first-child {
          text-indent: 0;
        }
    `};
    }
  }
`

const CaptionWrapper = styled(Sans)`
  flex: 1;

  ${media.sm`
    min-width: 50%;
  `};
`

const ImageSetCaption = styled.div`
  height: 100%;
  padding: ${space(2)}px;
  background: white;
  color: black;
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${media.sm`
    border-bottom: none;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ImageSetWrapper = styled(Box)`
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;

  ${ImageSetContainer} {
    display: flex;
    flex-direction: row-reverse;

    ${media.sm`
      flex-direction: column-reverse;
    `};
  }

  ${FullLabel} {
    position: relative;
    display: flex;
    flex-direction: column-reverse;
    flex: 1;
    left: auto;
    bottom: auto;
    max-width: inherit;
    border-radius: 0;
    box-shadow: none;

    &:hover {
      background-color: ${color("purple100")};
    }
    ${media.sm`
      flex-direction: row-reverse;

      ${LabelWrapper} {
        max-width: 50%;
      }
    `};
  }

  ${SlideshowTitle} {
    display: none;
  }

  ${SlideshowCta} {
    flex-direction: column;

    > div {
      padding: 0;
    }
  }

  ${ImgContainer} {
    flex: 3;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

    img {
      object-fit: cover;
      object-position: center;
      height: 100%;
    }

    ${media.sm`
      border-right: none;
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};
  }
`

const RelatedArticleWrapper = styled.div`
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${ArticlesWrapper} {
    margin-bottom: 0;
    margin-top: 0;
  }

  ${ArticleCardContainer} {
    border: none;
  }
`
