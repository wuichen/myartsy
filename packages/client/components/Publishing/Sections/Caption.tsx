import { garamond, unica } from "Assets/Fonts"
import { pMedia } from "Components/Helpers"
import { ArticleLayout, SectionLayout } from "Components/Publishing/Typings"
import React from "react"
import styled from "styled-components"

interface CaptionProps {
  caption: string
  color?: string
  index?: any
  layout?: ArticleLayout
  sectionLayout?: SectionLayout
}

interface FigcaptionProps {
  color?: string
  layout: ArticleLayout
  sectionLayout?: SectionLayout
}

export const Caption: React.SFC<CaptionProps> = props => {
  const { caption, children, color, layout, sectionLayout } = props

  const child = children || (
    <div dangerouslySetInnerHTML={{ __html: caption }} />
  )

  return (
    <CaptionContainer>
      <Figcaption layout={layout} sectionLayout={sectionLayout} color={color}>
        {child}
      </Figcaption>
    </CaptionContainer>
  )
}

export const CaptionContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 10px 0 10px 0;

  ${pMedia.xs`
    padding: 0px 10px;
  `};
`

// includes draft placeholder class for editable text in Writer
const Figcaption = styled.div<FigcaptionProps>`
  padding: ${props => (props.sectionLayout === "fillwidth" ? "0 10px;" : "0;")}
  width: 100%;
  word-break: break-word;

  & > p, p,
  .public-DraftEditorPlaceholder-root,
  .public-DraftStyleDefault-block {
    ${props => (props.layout === "classic" ? garamond("s15") : unica("s14"))}
    color: ${props =>
      props.color ? props.color : props.layout === "classic" ? "#666" : "#999"};
    margin: 0;
    display: inline;    
  }

  & > a, a {
    color: #999;
    &:hover {
      color: black;
    }
  }

  ${pMedia.xs`
    padding: 0px;
  `}
`
