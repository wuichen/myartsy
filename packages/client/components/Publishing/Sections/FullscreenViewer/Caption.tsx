import Colors from "Assets/Colors"
import { unica } from "Assets/Fonts"
import { pMedia } from "Components/Helpers"
import PropTypes from "prop-types"
import React from "react"
import styled, { StyledFunction } from "styled-components"
import { ArtworkCaption } from "../ArtworkCaption"

interface CaptionProps extends React.HTMLProps<HTMLDivElement> {
  section?: any
  total: number
  index: number
  open: boolean
}

interface CaptionOpenProps extends React.HTMLProps<HTMLDivElement> {
  open: boolean
}

export const Caption: React.SFC<CaptionProps> = props => {
  const { index, open, section, total } = props

  const isArtwork = section.type === "artwork"
  const indexText = `${index} of ${total}`

  return (
    <CaptionContainer>
      <CaptionTextContainer>
        <CaptionToggle open={open} />

        <CaptionText open={open}>
          {isArtwork ? (
            <ArtworkCaption artwork={section} isFullscreenCaption linked />
          ) : (
            <div
              dangerouslySetInnerHTML={{
                __html: section.caption,
              }}
            />
          )}
        </CaptionText>
      </CaptionTextContainer>

      <Index>{indexText}</Index>
    </CaptionContainer>
  )
}

const CaptionToggle: React.SFC<CaptionOpenProps> = (props, context) => {
  const toggleMessage = props.open ? "Hide" : "View Caption"

  return (
    <StyledCaptionToggle
      onClick={context.onToggleCaption}
      className="fullscreen-viewer__caption-toggle"
    >
      <span>{toggleMessage}</span>
    </StyledCaptionToggle>
  )
}

CaptionToggle.contextTypes = {
  onToggleCaption: PropTypes.func,
}

const StyledCaptionToggle = styled.div`
  display: none;
  ${pMedia.sm`
    ${unica("s14", "medium")}
    cursor: pointer;
    display: inline-block;
    span {
      border-bottom: 1px solid black;
    }
  `};
`

const CaptionDiv: StyledFunction<CaptionOpenProps> = styled.div

const CaptionText = CaptionDiv`
  ${unica("s16", "medium")}
  a {
    color: black;
  }
  ${props => pMedia.sm`
    ${unica("s14", "medium")}
    display: ${props.open ? "block" : "none"};
    margin-top: ${props.open ? "20px" : "0px"};
  `}
`

const Index = styled.div`
  margin-left: 20px;
  white-space: nowrap;
  ${unica("s16")};
  ${pMedia.sm`
    ${unica("s14")}
  `};
`

const CaptionContainer = styled.div`
  display: flex;
  justify-content: space-between;
  background-color: ${Colors.gray};
  padding: 30px 60px;

  p {
    margin: 0;
  }

  ${pMedia.sm`
    position: absolute;
    bottom: 0;
    width: 100%;
    max-width: 100vh;
    padding: 20px;
  `};
`

const CaptionTextContainer = styled.div`
  display: flex;
  flex-direction: column;
`
