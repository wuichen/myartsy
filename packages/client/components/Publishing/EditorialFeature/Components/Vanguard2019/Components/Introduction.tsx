import { Box, color, Flex, media, Sans, Serif } from "@artsy/palette"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { VanguardCredits } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Components/VanguardCredits"
import { Text } from "Components/Publishing/Sections/Text"
import { ArticleData } from "Components/Publishing/Typings"
import React from "react"
import styled from "styled-components"
import { Media } from "Utils/Responsive"
import { VanguardVideoHeader } from "./VideoHeader"

export const VanguardIntroduction: React.SFC<{
  article: ArticleData
  isMobile: boolean
}> = props => {
  const { isMobile } = props
  const { description } = props.article.series
  const { hero_section } = props.article
  const url = ((hero_section && hero_section.url) || "") as string
  const isVideo = url.includes("mp4")

  return (
    <IntroContainer>
      <Box minHeight="calc(100vw * 0.53)" pt={[100, 100, 50, 50]}>
        {isVideo && <VanguardVideoHeader url={url} />}

        <Media greaterThan="xs">
          <HeaderText
            pt={70}
            size={["12", "12", "14", "16"]}
            textAlign="center"
          >
            The Artsy
          </HeaderText>
        </Media>
      </Box>

      <Box background={color("white100")} pt={[80, 100, 150]}>
        <Box mx="auto" maxWidth={980} px={4}>
          <Flex flexDirection="column" alignItems="center" pb={50}>
            <Media greaterThanOrEqual="xl">
              <LargeTitle size="12" element="h1" textAlign="center" pb={1}>
                The Artists To Know Right Now
              </LargeTitle>
            </Media>

            <Media lessThan="xl">
              <Title
                size={["8", "10", "12", "12"]}
                element="h1"
                textAlign="center"
                pb={1}
              >
                The Artists To Know Right Now
              </Title>
            </Media>

            <Byline {...props} />

            {!isMobile && <VanguardCredits isMobile={isMobile} />}
          </Flex>

          <Box pb={12}>
            <Text layout="standard" html={description} width="800px" />
          </Box>
        </Box>
      </Box>
    </IntroContainer>
  )
}

const HeaderText = styled(Sans)`
  position: absolute;
  top: 0;
  width: 100%;
  z-index: -1;
  color: ${color("white100")};
  mix-blend-mode: difference;
  will-change: color;
`

const IntroContainer = styled(Box)`
  ${BylineContainer} {
    div::before {
      display: none;
    }
  }

  ${media.xs`
    ${BylineContainer} {
      flex-direction: column;
      align-items: center;
    }
  `}
`

const Title = styled(Serif)`
  text-transform: uppercase;
  line-height: 1em;
`

const LargeTitle = styled(Title)`
  font-size: 90px;
`
