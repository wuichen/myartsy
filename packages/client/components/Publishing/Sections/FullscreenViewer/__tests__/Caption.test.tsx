import { Images } from "Components/Publishing/Fixtures/Components"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { Caption } from "../Caption"

it("renders an artwork caption properly", () => {
  const caption = renderer
    .create(<Caption section={Images[0]} total={10} index={2} open />)
    .toJSON()
  expect(caption).toMatchSnapshot()
})

it("renders an image caption properly", () => {
  const caption = renderer
    .create(<Caption section={Images[1]} total={10} index={3} open />)
    .toJSON()
  expect(caption).toMatchSnapshot()
})

it("toggles the caption on mobile to hide", () => {
  const context = { onToggleCaption: jest.fn() }
  const caption = mount(
    <Caption section={Images[0]} total={10} index={2} open />,
    {
      childContextTypes: context,
      context,
    }
  )
  const node = caption.find(".fullscreen-viewer__caption-toggle").at(0)
  expect(node.text()).toBe("Hide")
  node.simulate("click")
  expect(context.onToggleCaption.mock.calls.length).toBeGreaterThan(0)
})

it("toggles the caption on mobile to view", () => {
  const context = { onToggleCaption: jest.fn() }
  const caption = mount(
    <Caption section={Images[0]} total={10} index={2} open={false} />,
    {
      childContextTypes: context,
      context,
    }
  )
  const node = caption.find(".fullscreen-viewer__caption-toggle").at(0)
  expect(node.text()).toBe("View Caption")
  node.simulate("click")
  expect(context.onToggleCaption.mock.calls.length).toBeGreaterThan(0)
})
