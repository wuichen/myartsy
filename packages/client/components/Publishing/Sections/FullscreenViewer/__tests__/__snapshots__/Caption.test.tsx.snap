// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an artwork caption properly 1`] = `
.c5 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 24px;
}

.c7 {
  margin-right: 30px;
}

.c8 {
  white-space: nowrap;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 .title,
.c4 .title a {
  font-style: italic;
}

.c2 {
  display: none;
}

.c3 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c3 a {
  color: black;
}

.c9 {
  margin-left: 20px;
  white-space: nowrap;
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #f8f8f8;
  padding: 30px 60px;
}

.c0 p {
  margin: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media screen and (min-width:40em) {
  .c5 {
    font-size: 16px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    line-height: 26px;
  }
}

@media (max-width:720px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:720px) {
  .c6.artist-name {
    margin-bottom: 5px;
  }
}

@media (max-width:720px) {
  .c2 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    cursor: pointer;
    display: inline-block;
  }

  .c2 span {
    border-bottom: 1px solid black;
  }
}

@media (max-width:720px) {
  .c3 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    display: block;
    margin-top: 20px;
  }
}

@media (max-width:720px) {
  .c9 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
  }
}

@media (max-width:720px) {
  .c0 {
    position: absolute;
    bottom: 0;
    width: 100%;
    max-width: 100vh;
    padding: 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="fullscreen-viewer__caption-toggle c2"
    >
      <span>
        Hide
      </span>
    </div>
    <div
      className="c3"
      open={true}
    >
      <div>
        <div
          className="c4 c5"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize={
            Array [
              "14px",
              "16px",
            ]
          }
        >
          <div
            className="c6"
          >
            <span
              className="c7"
            >
              <span
                className="c8"
              >
                <a
                  href="/artist/fernando-botero"
                >
                  Fernando Botero
                </a>
              </span>
            </span>
          </div>
          <div>
            <div
              className="c6"
            >
              <span
                className="title"
              >
                <a
                  href="/artwork/fernando-botero-nude-on-the-beach"
                >
                  Nude on the Beach
                </a>
              </span>
              <span>
                , 
              </span>
              <span
                className="date"
              >
                2000
              </span>
            </div>
            <div
              className="c6"
            >
              <a
                href="/gary-nader"
              >
                Gary Nader
              </a>
              <span>
                . 
              </span>
              <span
                className="credit"
              >
                Courtesy of Gary Nader
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c9"
  >
    2 of 10
  </div>
</div>
`;

exports[`renders an image caption properly 1`] = `
.c2 {
  display: none;
}

.c3 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c3 a {
  color: black;
}

.c4 {
  margin-left: 20px;
  white-space: nowrap;
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #f8f8f8;
  padding: 30px 60px;
}

.c0 p {
  margin: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width:720px) {
  .c2 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    cursor: pointer;
    display: inline-block;
  }

  .c2 span {
    border-bottom: 1px solid black;
  }
}

@media (max-width:720px) {
  .c3 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    display: block;
    margin-top: 20px;
  }
}

@media (max-width:720px) {
  .c4 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
  }
}

@media (max-width:720px) {
  .c0 {
    position: absolute;
    bottom: 0;
    width: 100%;
    max-width: 100vh;
    padding: 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="fullscreen-viewer__caption-toggle c2"
    >
      <span>
        Hide
      </span>
    </div>
    <div
      className="c3"
      open={true}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Photo by Adam Kuehl for Artsy.</p>",
          }
        }
      />
    </div>
  </div>
  <div
    className="c4"
  >
    3 of 10
  </div>
</div>
`;
